### PR TITLE
Move jellyfinclient folder contents up one level

### PR DESCRIPTION
**Pull request by LucaBernstein**
See #15 

I see that for the actively working developers on this project this merge would mean chaning their dev environment once (which could be annoying) but in my opinion it makes more sense to pull the `jellyfinclient` folder contents up one level.

I would be glad to hear your feedback and what you say to this idea 😃 

Thanks in advance,

Luca

### CODE REVIEW

**Comment by DeveloperOl** [comment]
I like this one. Or is there any specific reason, why you initially developed the jellyfinclient in a subdir @EraYaN ?

**Comment by EraYaN** [comment]
Well yes since react-native projects break when their containing folder is not the folder they were created in. Now it’s exported so it might work.

But please do test all builds first, Android, iOS, tvOS, Windows, Web, Electron etc.

**Review by DrPandemic — APPROVED** [review]
## web
I was able to build web.

## android
I'm pretty sure master doesn't build master right now. There's many scripts that don't have exec permissions and typo in path. With some patches I was able to build and run on Android. [I'll open a PR for that](https://github.com/jellyfin/jellyfin-react-client/pull/18).

## electron
My system is kinda borked so I'm unable to build the Electron app. Something about QT being broken... not sure. If someone else could test it, it would be nice.

## UWP
I can't test UWP.

## iOS
I also don't have a setup for ios, but I'm confident that since I was able to build android, ios should be good.

## tvOS?
I didn't see any script for tvOS. Do we have something special for that?

**Comment by LucaBernstein** [comment]
I don't see why it should not build. The only thing I could imagine is that if other repos would rely on this one to have a fixed folder structure that they would need to be adapted as well. @EraYaN Do you know if this is the case?

I ran the builds (releases) on Windows by using `yarn release` and the success was as follows:

## Android (SUCCESS)

I was able to build for Android after simply running `yarn install` again.

```
$ yarn run android:release && yarn run electron:release && yarn run ios:release && yarn run web:release && yarn run windows:release
$ cd android && ./gradlew assembleRelease

> Configure project :app 
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> Task :app:bundleReleaseJsAndAssets 
warning: the transform cache was reset.
Loading dependency graph, done.
bundle: Writing bundle output to: C:\Java-Dev\GitHub_Forks\jellyfin-react-client\android\app\build\generated\assets\react\release\index.android.bundle
bundle: Done writing bundle output
bundle: Copying 1 asset files
bundle: Done copying assets


BUILD SUCCESSFUL in 1m 46s
55 actionable tasks: 2 executed, 53 up-to-date
```

## Electron (SUCCESS)

```
$ yarn run preelectron
$ cross-env-shell ELECTRON=1 PUBLIC_URL=. "yarn run web:release && shx cp index.electron.js build/index.js && echo {} > build/package.json"
$ yarn run icon-gen
$ icon-gen -i resources/icon.svg -o public && shx cp public/favicon-228.png public/app.png
$ node scripts/build.js
Loading electron webpack config...
Creating an optimized production build...
Compiled with warnings.

./src/actions/LoginAction.js
  Line 17:  'loginFailed' is defined but never used  no-unused-vars

./node_modules/write-file-atomic/index.js
Module not found: Can't resolve 'worker_threads' in 'C:\Java-Dev\GitHub_Forks\jellyfin-react-client\node_modules\write-file-atomic'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

File sizes after gzip:

  183.85 KB (+51 B)  build\static\js\1.2c66e3a4.chunk.js
  3.94 KB (-13 B)    build\static\js\main.f9c584fa.chunk.js
  763 B              build\static\js\runtime~main.4a686d48.js

The project was built assuming it is hosted at ./.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

Find out more about deployment here:

  http://bit.ly/CRA-deploy
```

## iOS (FAILED)

Though I would guess this is due to my Windows machine. If someone could run that on a Mac device that should work imo, maybe one could test that (has that ever worked before?)

```
react-native bundle --platform=ios

error: option '--entry-file' missing

Error: error: option '--entry-file' missing
    at options.forEach.opt (C:\Java-Dev\GitHub_Forks\jellyfin-react-client\node_modules\react-native\local-cli\util\assertRequiredOptions.js:30:13)
    at Array.forEach (<anonymous>)
    at assertRequiredOptions (C:\Java-Dev\GitHub_Forks\jellyfin-react-client\node_modules\react-native\local-cli\util\assertRequiredOptions.js:19:11)
    at Promise.resolve.then (C:\Java-Dev\GitHub_Forks\jellyfin-react-client\node_modules\react-native\local-cli\cliEntry.js:116:7)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Web (SUCCESS)

```
yarn run web:release
$ yarn run icon-gen
$ icon-gen -i resources/icon.svg -o public && shx cp public/favicon-228.png public/app.png
$ node scripts/build.js
Loading main webpack config...
Creating an optimized production build...
Compiled with warnings.

./src/actions/LoginAction.js
  Line 17:  'loginFailed' is defined but never used  no-unused-vars

./node_modules/write-file-atomic/index.js
Module not found: Can't resolve 'worker_threads' in 'C:\Java-Dev\GitHub_Forks\jellyfin-react-client\node_modules\write-file-atomic'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

File sizes after gzip:

  289.88 KB (+106.03 KB)  build\static\js\1.419d0892.chunk.js
  3.87 KB (-75 B)         build\static\js\main.a9feba66.chunk.js
  763 B                   build\static\js\runtime~main.229c360f.js

The project was built assuming it is hosted at the server root.
You can control this with the homepage field in your package.json.
For example, add this to build it for GitHub Pages:

  "homepage" : "http://myname.github.io/myapp",

The build folder is ready to be deployed.
You may serve it with a static server:

  yarn global add serve
  serve -s build

Find out more about deployment here:

  http://bit.ly/CRA-deploy

Done in 67.21s.
```

## Windows (SUCCESS)

```
yarn run windows:release
$ react-native bundle --platform=windows --entry-file index.js --assets-dest windows/jellyfinclient/ReactAssets --bundle-output windows/jellyfinclient/ReactAssets/index.windows.bundle
--dev false
Loading dependency graph, done.
bundle: Writing bundle output to: windows/jellyfinclient/ReactAssets/index.windows.bundle
bundle: Done writing bundle output
bundle: Copying 1 asset files
bundle: Done copying assets
Done in 81.25s.
```

**Comment by anthonylavado** [comment]
I can test on a Mac, but as the documentation guy, I have no experience in building this. What commands do I need to run?

**Comment by DrPandemic** [comment]
You'll need `yarn run ios`. I guess you'll need xcode and ios dev tools installed

**Comment by EraYaN** [comment]
Please recreate this PR, since it only moves files, a rewind, rebase and commit en force push should do the trick.

File: yarn.lock [hunk 1]
```diff
@@ -5,14 +5,12 @@
 "@babel/code-frame@7.0.0", "@babel/code-frame@^7.0.0", "@babel/code-frame@^7.0.0-beta.35":
   version "7.0.0"
   resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.0.0.tgz#06e2ab19bdb535385559aabb5ba59729482800f8"
-  integrity sha512-OfC2uemaknXr87bdLUkWog7nYuliM9Ij5HUcajsVcMCpQrcLmtxRbVFTIqmcSkSeYRBFBRxs2FiUqFJDLdiebA==
```
Comment (oddstr13) [inline yarn.lock]: Any particular reason for why all the integrity checks are all removed?

**Review by oddstr13 — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 1 above)
Comment (LucaBernstein) [reply yarn.lock]: No, maybe this was an error in my yarn setup. You are right, we should keep this clean here.

**Review by oddstr13 — APPROVED** [review]

**Review by EraYaN — APPROVED** [review]

**Pull request merged**